src/providers/openstack/configdrive.rs

```diff
@@ -0,0 +1,167 @@
+//! configdrive metadata fetcher for OpenStack
+//! reference: https://docs.openstack.org/nova/latest/user/metadata.html
+
+use std::collections::HashMap;
+use std::fs::File;
+use std::io::Read;
+use std::path::{Path, PathBuf};
+
+use openssh_keys::PublicKey;
+use slog_scope::{error, warn};
+use tempfile::TempDir;
+
+use crate::errors::*;
+use crate::network;
+use crate::providers::MetadataProvider;
+
+const CONFIG_DRIVE_LABEL: &str = "config-2";
+
+/// OpenStack config-drive.
+#[derive(Debug)]
+pub struct OpenstackConfigDrive {
+    /// Path to the top directory of the mounted config-drive.
+    drive_path: PathBuf,
+    /// Temporary directory for own mountpoint (if any).
+    temp_dir: Option<TempDir>,
+}
+
+impl OpenstackConfigDrive {
+    /// Try to build a new provider client.
+    ///
+    /// This internally tries to mount (and own) the config-drive.
+    pub fn try_new() -> Result<Self> {
+        // Short-circuit if the config-drive is already mounted.
+        let path = Path::new("/media/ConfigDrive/openstack/latest/");
```

Comment:
This path is copied from https://github.com/coreos/afterburn/blob/master/src/providers/cloudstack/configdrive.rs#L34, not entirely sure about the auto mount point of the config-drive. I think I might be missing some context on how the mount point is set up, the only documentation I found is https://coreos.com/os/docs/latest/cloud-config-locations.html

Reply:
I would be surprised if FCOS/RHCOS shipped anything to do this by default, but Flatcar might.

Reply:
It should be fine to drop this part. As long as all entities are mounting the config-drive read-only, it should be ok to have it mounted multiple time by different owners (each one handling the mount lifecycle on their own).